storage/innobase/log/log0recv.cc

```diff
@@ -3627,15 +3633,62 @@ static void recv_recovery_begin(log_t &log, lsn_t *contiguous_lsn,
 
   bool finished = false;
 
+  static int redo_applied = 0;
+  static int redo_applied_total = 0;
+
+
+
   while (!finished) {
     lsn_t end_lsn = start_lsn + RECV_SCAN_SIZE;
 
     recv_read_log_seg(log, log.buf, start_lsn, end_lsn);
 
+    bool flushed = false;
     finished = recv_scan_log_recs(log, max_mem, log.buf, RECV_SCAN_SIZE,
                                   checkpoint_lsn, start_lsn, contiguous_lsn,
-                                  &log.scanned_lsn, to_lsn);
+                                  &log.scanned_lsn, to_lsn, &flushed);
+
+    redo_applied += RECV_SCAN_SIZE;
+    if (flushed) {
+      redo_applied_total += redo_applied;
+
+      auto pct = ((double)redo_applied / srv_log_file_size) * 100;
+      auto total_applied_per =
+          ((double)redo_applied_total / (double)srv_log_file_size) * 100;
+
+      ib::info() << "redo applied in last iteration:" << pct
+                 << "%\n total redo applied:" << total_applied_per << "%";
+
+      /* find availabe free memory */
+      auto free_memory =
+          (ulint)(double(sysconf(_SC_PHYS_PAGES) * sysconf(_SC_PAGESIZE) *
+                         xtrabackup_free_memory_per) /
+                  100);
 
```

Comment:
Please create a separate function for this.

Add code for macos, with ifdefs.
See this for free memory calculation:
https://stackoverflow.com/questions/6094444/how-can-i-programmatically-check-free-system-memory-on-mac-like-the-activity-mon/6095158